docs/tutorial/file_handling.md

```diff
@@ -0,0 +1,471 @@
+
+This is how you can create a file in Shellfish:
+
+@:select(api-style)
+
+@:choice(syntax)
+
+```scala mdoc:compile-only
+import io.chrisdavenport.shellfish.syntax.path.*
+import fs2.io.file.Path
+import cats.effect.{IO, IOApp}
+
+object Creating extends IOApp.Simple:
+
+  val filePath = Path("path/to/your/desired/creation/NiceScript.scala")
+
+  def run: IO[Unit] =
+    for
+      _ <- filePath.createFile
+      created <- filePath.exists
+      _ <- IO.println(s"File created? $created")
+    yield ()
+
+end Creating
+```
+
+@:choice(static)
+
+```scala mdoc:compile-only
+import io.chrisdavenport.shellfish.FilesOs
+import fs2.io.file.Path
+import cats.effect.{IO, IOApp}
+
+object Creating extends IOApp.Simple:
+
+  val filePath = Path("path/to/your/desired/creation/NiceScript.scala")
+
+  def run: IO[Unit] =
+    for
+      _ <- FilesOs.createFile(filePath)
+      created <- FilesOs.exists(filePath)
+      _ <- IO.println(s"File created? $created")
+    yield ()
+
+end Creating
+```
+
+@:@
+
+Here, we are first creating the file using the `createFile` method and then checking its existent with the `exists` method.
+
+**Important:** The `createFile` and `createDirectory` methods will only work if the parent directory already exists, otherwise it will fail so be careful with that. Because of that, if you want to create the parent directories, as well you should use the `createDirectories` method like this:
+
+@:select(api-style)
+
+@:choice(syntax)
+
+```scala mdoc:compile-only
+import io.chrisdavenport.shellfish.syntax.path.*
+import fs2.io.file.Path
+import cats.effect.{IO, IOApp}
+
+object Creating extends IOApp.Simple:
+
+  val emptyDirectories = Path("create/me/first")
+
+  def run: IO[Unit] =
+    emptyDirectories.createDirectories >> Path("now_i_can_be_created.fs").createFile
+
+end Creating
+```
+
+@:choice(static)
+
+```scala mdoc:compile-only
+import io.chrisdavenport.shellfish.FilesOs
+import fs2.io.file.Path
+import cats.effect.{IO, IOApp}
+
+object Creating extends IOApp.Simple:
+
+  val emptyDirectories = Path("create/me/first")
+
+  def run: IO[Unit] =
+    FilesOs.createDirectories(emptyDirectories) >> 
+      FilesOs.createFile(Path("now_i_can_be_created.fs"))
+
+end Creating
+```
+
+@:@
+
+### Exercise
+
+Imagine that you need to create a lot of nested files in a directory that also needs the creation of the subsequent folders. For that reason, you need to implement a function that creates both the file and the parent directories. Can you implement it? How?
+
+@:select(api-style)
+
+@:choice(syntax)
+
+```scala
+extension (path: Path) def createFileAndDirectories: IO[Unit] = ???
+```
+
+@:choice(static)
+
+```scala
+def createFileAndDirectories(path: Path): IO[Unit] = ???
+```
+
+@:@
+
+[See possible implementation](https://gist.github.com/00c59408740e8b52838ff07fc9154c12.git)
```

Comment:
Does Laika support spoilers like Github Flavored markdown? If so it's better to inline this in the doc:
```markdown

<details>
  <summary>Possible implementation</summary>

import io.chrisdavenport.shellfish.syntax.path.*

extension (path: Path) 
  def createFileAndDirectories: IO[Unit] = 
    path.parent match
      case Some(dirs) => dirs.createDirectories >> path.fileName.createFile
      case None       => path.createFile
</details>
```

Reply:
I also don't like pointing at personal gists with the solutions. We need to find a better way. At the moment Laika doesn't seem to be supporting this custom html directives (AFAIK).

I'll ask you to give readers both the solutions anyway.

Reply:
Laika also doesn't support spoilers using the `>!` operators, as it renders like this: 

>! Hello im a spoiler

Reply:
Yea I think that's a fine solution for now.

Reply:
Yeah, that could work